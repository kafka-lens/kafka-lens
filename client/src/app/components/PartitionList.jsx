import React from 'react';
import Partition from './Partition.jsx';

const PartitionList = (props) => {
  const partitionsArray = [];
  const { numberOfPartitions } = props.topicInfo;

  for (let i = 0; i < numberOfPartitions; i++) {
    partitionsArray.push(
      <Partition
        key={i}
        id={i}
        showMessages={props.showMessages}
        topicName={props.topicInfo.topicName}
      />,
    );
  }

  return <div className="partition-list">{partitionsArray}</div>;
};

export default PartitionList;
