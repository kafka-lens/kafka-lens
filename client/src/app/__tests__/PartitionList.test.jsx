import React from 'react';
import { shallow } from 'enzyme';

import PartitionList from '../components/PartitionList';

describe('PartitionList unit tests', () => {
  let wrapper;
  const props = {
    topicInfo: {
      numberOfPartitions: 8,
    },
  };
  beforeAll(() => {
    wrapper = shallow(<PartitionList {...props} />);
  });

  it('Should render without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render a Partition component', () => {
    expect(wrapper.exists('Partition')).toBe(true);
  });
});
