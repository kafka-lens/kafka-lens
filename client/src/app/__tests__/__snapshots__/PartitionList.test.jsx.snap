// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PartitionList unit tests Should render without errors 1`] = `
<div
  className="partition-list"
>
  <Partition
    id={0}
    key="0"
  />
  <Partition
    id={1}
    key="1"
  />
  <Partition
    id={2}
    key="2"
  />
  <Partition
    id={3}
    key="3"
  />
  <Partition
    id={4}
    key="4"
  />
  <Partition
    id={5}
    key="5"
  />
  <Partition
    id={6}
    key="6"
  />
  <Partition
    id={7}
    key="7"
  />
</div>
`;
